server/src/database/typeorm/typeorm.service.ts

```diff
@@ -25,24 +25,28 @@ export class TypeORMService implements OnModuleInit, OnModuleDestroy {
    * @returns Promise<DataSource | undefined>
    */
   public async connectToDataSource(
-    dataSource: DataSourceEntity,
+    dataSource: Partial<DataSourceEntity>,
   ): Promise<DataSource | undefined> {
-    if (this.dataSources.has(dataSource.id)) {
+    if (dataSource.id && this.dataSources.has(dataSource.id)) {
       return this.dataSources.get(dataSource.id);
     }
 
     const schema = dataSource.schema;
 
+    const debugMode = this.environmentService.isDebugMode();
+
     const workspaceDataSource = new DataSource({
       url: dataSource.url ?? this.environmentService.getPGDatabaseUrl(),
       type: 'postgres',
-      logging: ['query'],
+      logging: debugMode ? ['query'] : ['error'],
       schema,
     });
 
     await workspaceDataSource.initialize();
 
-    this.dataSources.set(dataSource.id, workspaceDataSource);
+    if (dataSource.id) {
```

Comment:
Can we check this before instantiating a DataSource?

Reply:
in fact we need that just to cache the datasource, so it needs to be after instantiation